apps/central-scan/frontend/src/components/delete_batch_modal.tsx

```diff
@@ -0,0 +1,62 @@
+import { Button, Modal, Prose, Text } from '@votingworks/ui';
+import React from 'react';
+import * as api from '../api';
+
+export interface Props {
+  batchId: string;
+  batchLabel: string;
+  onClose: () => void;
+}
+
+/**
+ * Provides a modal for confirming the deletion of a scanned ballot batch.
+ */
+export function DeleteBatchModal({
+  batchId,
+  batchLabel,
+  onClose,
+}: Props): JSX.Element {
+  const deleteBatchMutation = api.deleteBatch.useMutation();
+
+  async function doDeleteBatch() {
+    try {
+      // await the mutation to ensure the modal is closed only after the mutation
+      // is complete and the batch is deleted.
+      await deleteBatchMutation.mutateAsync({ batchId });
+      onClose();
```

Comment:
The pattern I've been recommending is to do `mutate` with an `onSuccess` handler for this. Any reason you would prefer `mutateAsync` with an empty `catch`? I know it's a bit more readable to be able to use `await`, but I'm interested in having a consistent approach as much as possible to reduce cognitive overhead.

Reply:
I had played around with a couple different options and never landed on that for some reason, but it does seem better. Going with the callback.